platform/javascript/js/engine/engine.js

```diff
@@ -34,30 +34,28 @@ const Engine = (function() {
 		this.onExecute = null;
 		this.onExit = null;
 		this.persistentPaths = ['/userfs'];
-	};
+	}
 
-	Engine.prototype.init = /** @param {string=} basePath */ function(basePath) {
+	Engine.prototype.init = /** @param {string=} basePath */ function (basePath) {
 		if (initPromise) {
 			return initPromise;
 		}
 		if (loadPromise == null) {
 			if (!basePath) {
-				initPromise = Promise.reject(new Error("A base path must be provided when calling `init` and the engine is not loaded."));
+				initPromise = Promise.reject(new Error('A base path must be provided when calling `init` and the engine is not loaded.'));
 				return initPromise;
 			}
 			load(basePath);
 		}
-		var config = {};
-		if (typeof stdout === 'function')
-			config.print = stdout;
-		if (typeof stderr === 'function')
-			config.printErr = stderr;
-		var me = this;
-		initPromise = new Promise(function(resolve, reject) {
+		let config = {};
+		if (typeof stdout === 'function') config.print = stdout;
+		if (typeof stderr === 'function') config.printErr = stderr;
```

Comment:
Is the style either braces or single-line `if`s?

Reply:
Yes: https://eslint.org/docs/rules/curly
Should be: `curly: ["error", "multi"]`.
I'm okay to change it to `curly: "error"` (i.e. always require curly, even for oneliner).

Reply:
I leave it up to you, I don't haven't a strong opinion for JS. But yeah keeping it consistent with C++ might be good if it doesn't go against typical JS style recommendations.